staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go

```diff
@@ -134,3 +134,43 @@ func TestKindForGroupVersionKinds(t *testing.T) {
 		}
 	}
 }
+
+func TestParseKindArg(t *testing.T) {
+	tests := []struct {
+		input string
+		gvk   *GroupVersionKind
+		gk    GroupKind
+	}{
+		{input: "Pod", gk: GroupKind{Kind: "Pod"}},
+		{input: ".apps", gk: GroupKind{Group: "apps"}},
+		{input: "Pod.", gk: GroupKind{Kind: "Pod"}},
+		{input: "Pod.apps", gk: GroupKind{Group: "apps", Kind: "Pod"}},
+		{input: "Pod.v1.apps", gvk: &GroupVersionKind{Group: "apps", Version: "v1", Kind: "Pod"}, gk: GroupKind{Group: "v1.apps", Kind: "Pod"}},
+	}
+	for i, test := range tests {
+		gvk, gk := ParseKindArg(test.input)
+		if (gvk != nil && test.gvk == nil) || (gvk == nil && test.gvk != nil) || (test.gvk != nil && *gvk != *test.gvk) {
+			t.Errorf("%d: expected output: %#v, got: %#v", i, test.gvk, gvk)
```

Comment:
Please use the subtest construct: https://blog.golang.org/subtests

Reply:
@deads2k Done